runtime/near-vm-runner/src/wasmer2_runner.rs

```diff
@@ -284,53 +272,70 @@ fn run_method(module: &Module, import: &ImportObject, method_name: &str) -> Resu
     Ok(())
 }
 
-pub(crate) fn compile_wasmer1_module(code: &[u8]) -> bool {
-    let store = default_wasmer1_store();
+pub(crate) fn compile_wasmer2_module(code: &[u8]) -> bool {
+    let store = default_wasmer2_store();
     Module::new(&store, code).is_ok()
 }
 
+#[derive(Hash, PartialEq, Debug)]
+#[allow(unused)]
+enum WasmerEngine {
+    Universal = 1,
+    StaticLib = 2,
+    DynamicLib = 3,
+}
+
+#[derive(Hash, PartialEq, Debug)]
+#[allow(unused)]
+enum WasmerCompiler {
+    Singlepass = 1,
+    Cranelift = 2,
+    Llvm = 3,
+}
+
 #[derive(Hash)]
-struct Wasmer1Config {
+struct Wasmer2Config {
     seed: i32,
-    use_cranelift: bool,
-    use_native_engine: bool,
+    engine: WasmerEngine,
+    compiler: WasmerCompiler,
 }
 
-impl Wasmer1Config {
+impl Wasmer2Config {
     fn config_hash(self: Self) -> u64 {
-        let mut s = DefaultHasher::new();
+        let mut s = StableHasher::new();
         self.hash(&mut s);
         s.finish()
     }
 }
 
-const WASMER1_CONFIG: Wasmer1Config =
-    Wasmer1Config { seed: 53, use_cranelift: false, use_native_engine: false };
+// We use following scheme for the bits forming seed:
+//  kind << 10, kind is 1 for Wasmer
+//  major version << 6
+//  minor version
+const WASMER2_CONFIG: Wasmer2Config = Wasmer2Config {
+    seed: (1 << 10) | (2 << 6) | 0,
+    engine: WasmerEngine::Universal,
+    compiler: WasmerCompiler::Singlepass,
+};
 
-pub(crate) fn wasmer1_vm_hash() -> u64 {
-    WASMER1_CONFIG.config_hash()
+pub(crate) fn wasmer2_vm_hash() -> u64 {
+    WASMER2_CONFIG.config_hash()
 }
 
-pub(crate) fn default_wasmer1_store() -> Store {
-    if WASMER1_CONFIG.use_native_engine {
-        let engine = if WASMER1_CONFIG.use_cranelift {
-            wasmer_engine_native::Native::new(wasmer_compiler_cranelift::Cranelift::default())
-                .engine()
-        } else {
-            wasmer_engine_native::Native::new(wasmer_compiler_singlepass::Singlepass::default())
-                .engine()
-        };
-        Store::new(&engine)
-    } else {
-        let engine = JIT::new(Singlepass::default()).engine();
-        Store::new(&engine)
-    }
+pub(crate) fn default_wasmer2_store() -> Store {
```

Comment:
though: unrelated to pr, but it's probably better to spell this as `wasmer::Store` to not confuse with the store we already have: https://github.com/near/nearcore/blob/3a5aa9132145d6411f1b9b30abc98a6a7a609ea5/core/store/src/lib.rs#L48-L52